Tests/UI/New UI/BillingForm/View/ExpiryDateViewTests.swift

```diff
@@ -89,4 +89,139 @@ class ExpiryDateViewTests: XCTestCase {
         view.updateExpiryDate(to: "Test")
         XCTAssertFalse(view.style?.error?.isHidden ?? true)
     }
+
+  func testValidFirstDigitInputWith0() {
+    let input = "0"
+    let textField = UITextField()
+    textField.text = ""
+
+    let shouldContinueAdding = view.textField(textField, shouldChangeCharactersIn: NSRange(location: 0, length: 0), replacementString: input)
+    if shouldContinueAdding {
+      textField.text?.append(input)
+    }
+
+    XCTAssertTrue(shouldContinueAdding)
+    XCTAssertEqual(textField.text, "0")
+  }
+
+  func testInvalidSecondDigitInputWith0() {
+    let input = "0"
+    let textField = UITextField()
+    textField.text = "0"
+
+    let shouldContinueAdding = view.textField(textField, shouldChangeCharactersIn: NSRange(location: 1, length: 0), replacementString: input)
+    if shouldContinueAdding {
+      textField.text?.append(input)
+    }
+
+    XCTAssertFalse(shouldContinueAdding)
+    XCTAssertEqual(textField.text, "0")
+  }
+
+  func testValidSecondDigitInputWith3() {
+    let textField = UITextField()
+    textField.text = ""
+    let input = "3"
+
+    let shouldContinueAdding = view.textField(textField, shouldChangeCharactersIn: NSRange(location: 0, length: 0), replacementString: input)
+    if shouldContinueAdding {
+      textField.text?.append(input)
+    }
+
+    XCTAssertTrue(shouldContinueAdding)
+    XCTAssertEqual(textField.text, "03")
```

Comment:
```suggestion
    let textField = UITextField()
    textField.text = ""
    let input = "3"

    let shouldContinueAdding = view.textField(textField, shouldChangeCharactersIn: NSRange(location: 0, length: 0), replacementString: input)
    XCTAssertTrue(shouldContinueAdding)
    
    textField.text?.append(input)
    XCTAssertEqual(textField.text, "03")
```

Follow up from previous comment as this test has slightly different behaviour. In this case the reading would be
**Given** I have this expected input and prepared textField

**When** I try to add the input to the textField
**Then** I am allowed to

**When** I do add it
**Then** input is as expected

Reply:
if `shouldContinueAdding` is `false` then `textField.text` must not append the input.

Reply:
Yes, but this is a unit test, not live code. This is expectation of behaviour, not "if this, then that, otherwise something else".

Reply:
Updated